src/CMakeLists.txt

```
@@ -19,3 +19,23 @@ if (CMAKE_CXX_COMPILER)
       target_link_libraries (mongoc-cxx-check PRIVATE mongoc_shared)
    endif ()
 endif ()
+
+# public-header-warnings
+add_executable(public-header-warnings EXCLUDE_FROM_ALL ${CMAKE_CURRENT_SOURCE_DIR}/c-check.c)
+
+target_link_libraries(public-header-warnings PRIVATE mongoc_static)
```

```suggestion

if (TARGET mongoc_static)
   target_link_libraries(public-header-warnings PRIVATE mongoc_static)
else ()
   target_link_libraries(public-header-warnings PRIVATE mongoc_shared)
endif ()
```

Suggest checking if `mongoc_static` target is defined first. The static libraries may not be defined if CMake is configured with `ENABLE_STATIC=OFF`.